RevenueCatUI/CustomerCenter/Views/CustomerCenterCompletionHandler.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CustomerCenterCompletionHandler.swift
+//
+//  Created by Cesar de la Vega on 6/7/24.
+
+import Foundation
+import SwiftUI
+
+final class CustomerCenterCompletionHandler: ObservableObject {
```

Comment:
I've been wondering whether it would make more sense to call this `CustomerCenterActionHandler`. Then, we would notify whenever an action is performed.

My main concern with this approach is that it moves away from what we have in paywalls, where we have individual callbacks for many actions like purchase/restore started/failed/completed... But I feel since there are going to be more actions here, it makes more sense to do something like that. Wdyt?

Reply:
That's a good point... Right, I think at some point we will have actions that are no completion actions like `feedbackSurveyStarted` so we want this to be an action handler. I honestly prefer just having this handler with a bunch of different events that can be handled in a switch statement than a lot of functions like we do in paywalls

It is true that adding a new event to a sealed class is technically a breaking change, but I think it's assumable. It is also true that it's not possible to modify an existing case in an enum (and add more properties) without creating a new event, but that's also an issue with independent functions.

In summary I find it cleaner to use sealed class when we are going to have a lot of potential events.